app/redux/MarketSaga.js

```diff
@@ -68,9 +69,12 @@ export function* fetchOpenOrders(set_user_action) {
         const state = yield call([db_api, db_api.exec], 'get_open_orders', [username]);
         yield put(MarketReducer.actions.receiveOpenOrders(state));
 
-       const [account] = yield call(Apis.db_api, 'get_accounts', [username])
-       yield put(MarketReducer.actions.receiveAccount({ account }))
-
+       let [account] = yield call(Apis.db_api, 'get_accounts', [username])
+       if(account) {
+           account = fromJS(account)
+           yield put(MarketReducer.actions.receiveAccount({ account }))
+           yield put(g.actions.receiveAccount({ account }))
```

Comment:
Is `yield put(g.actions.receiveAccount({ account }))` required for the witness page fix? Is there possibly a more relevant place to put this line, just in case `fetchOpenOrders` is no longer called on every page load in the future?